### ISSUE TITLE
A complex generator of a lot of dates.

### ISSUE THREAD

**Issue by lk-geimfari** [lk-geimfari/mimesis#497]
Sometimes we need to generate a lot of `datetime` objects and at this moment we have not a good solution for this, so I suggest something like this:

```python
import random

from datetime import datetime, timedelta



def bulk_create_datetimes(date_start, date_end, **kwargs):
    """Bulk create datetime objects

    :param date_start: Begin of the range.
    :param date_end: End of the range.
    :param kwargs: Keyword arguments for datetime.timedelta
    :return: List of datetime objects
    """

    dt_objects = []

    assert date_start and date_end

    while date_start <= date_end:
        date_start += timedelta(**kwargs)
        dt_objects.append(date_start)

    return dt_objects
```

How it works:

```python
>>> week_ago = datetime.now() - timedelta(days=7)
>>> now = datetime.now()
>>> datetimes = bulk_create_datetimes(week_ago, now, minutes=60)
>>> for i in datetimes:
....    print(i)
```
Result:
```
2018-08-21 12:58:55.687991
2018-08-21 13:58:55.687991
2018-08-21 14:58:55.687991
2018-08-21 15:58:55.687991
2018-08-21 16:58:55.687991
...
2018-08-28 12:58:55.687991
```

So, I think that it's pretty simple to understand and there is no need for comments.

That gives extreme flexibility in generating dates because a user can generate dates with any difference between them (including seconds, minutes and etc.).

**Comment by lk-geimfari** [comment]
@sobolevn Your comment and criticism, please. Maybe we can do it the more elegant way, what do you think?

**Comment by e-vandenberg** [comment]
I think it may be more useful for a param to determine the number of datetimes returned, rather than the interval between them. Some simple math might be added to achieve this.

**Comment by e-vandenberg** [comment]
For example, a user might want to generate 50 datetimes. In the implementation above, it is not very simple to do this.

**Comment by lk-geimfari** [comment]
@e-vandenberg Actually we just need to do this:

```python
>>> two_mo_ago = now - timedelta(days=60)
>>> datetimes = bulk_create_datetimes(diapason=(two_mo_ago, now), days=1)
```

**Comment by lk-geimfari** [comment]
Maybe I've misunderstood you, so can you please show me your vision?

**Comment by e-vandenberg** [comment]
You're right, that also works quite well. 

What I had in mind looked something like this:

```python
>>> datetimes = bulk_create_datetimes(diapason=(two_mo_ago, now), num_results=50)
```

It's up to you to decide which you think is more useful.

**Comment by lk-geimfari** [comment]
@e-vandenberg Can you, please, illustrate to me how it will work. For example user have this range:

```python
end = datetime.now()
start = end - timedelta(days=300)
```
And this code:

```
datetimes = bulk_create_datetimes(diapason=(start, end), num_results=20)
```

What should be returned? Only the first 20 elements?

**Comment by e-vandenberg** [comment]
@lk-geimfari Below I have implemented my idea:

```python
from datetime import datetime, timedelta

def bulk_create_datetimes(diapason=(), *args, **kwargs):
    dt_objects = []
    # Maybe that's better to raise an exception 
    # instead of using default range?
    try:
        start, end = diapason
    except ValueError:
        end = datetime.now()
        start = end - timedelta(days=7)

    num_results = kwargs["num_results"]
    
    # dividend used to space out datetimes evenly
    dividend = (end - start)/num_results

    for i in range(num_results):
        start += dividend
        dt_objects.append(start)
    return dt_objects

```

Usage:

```python
>>> end = datetime.now()
>>> start = end - timedelta(days=300)
>>> datetimes = bulk_create_datetimes(diapason=(start, end), num_results=20)
>>> len(datetimes) # this is just for proof that there are 20 elements
20
>>> for i in datetimes:
....    print(i)
2017-11-17 17:32:13.629556
2017-12-02 17:32:13.629556
...
2018-08-29 17:32:13.629556

```

As seen above, the list `datetimes` is filled with `num_results` (in this case 20) evenly separated datetimes, that will range from `start` to `end`. The separation between the datetimes is found by executing `(end - start)/num_results` as seen in the code above. 

In the examlpe provided, you can see that there are 20 datetimes returned while also keeping the range of dates customizeable (beginning 300 days ago at `start` and ending today at `end`).

**Comment by lk-geimfari** [comment]
@e-vandenberg The problems with your solution that it does not work without `num_results`.

**Comment by lk-geimfari** [comment]
Also, the `datetimes` in the example above looks too chaotic, seems like values don't matter. In this case, it's better to just randomly choice values from the list.

Like this:

```
import random

from datetime import datetime, timedelta


def bulk_create_datetimes(diapason=(), number=None, *args, **kwargs):
    dt_objects = []
    try:
        start, end = diapason
    except ValueError:
        end = datetime.now()
        start = end - timedelta(days=7)

    while start <= end:
        start += timedelta(*args, **kwargs)
        dt_objects.append(start)

    if number and len(dt_objects) > number:
        random.shuffle(dt_objects)
        return dt_objects[number]

    return dt_objects

```

**Comment by lk-geimfari** [comment]
I've discussed this feature in Python.ru community in Slack and most people think that the `number` of values does not make sense here.

**Comment by e-vandenberg** [comment]
I thought it could be of use for people who want a specified number of dates generated, but I also see how you might think it might be redundant. 

I'll leave it to you to decide whether or not to implement the `number_of_results` param.

**Comment by lk-geimfari** [comment]
Fixed in #539
